src/Common/src/System/Text/OSEncoding.Windows.cs

```diff
@@ -11,11 +11,12 @@ namespace System.Text
     internal sealed class OSEncoding : Encoding
     {
         private readonly int _codePage;
-        private string _encodingName;
+        private readonly string _encodingName;
 
         internal OSEncoding(int codePage) : base(codePage)
         {
             _codePage = codePage;
+            _encodingName = "Codepage - " + _codePage.ToString();
```

Comment:
It seems like this was intentionally Lazy initialized, maybe to avoid allocating the string when not needed, why not just make `_encodingName` nullable?

Reply:
ok